### PR TITLE
Don't evaluate entities if no entities present in test data

### PR DESCRIPTION
**Pull request by JEM-Mosig**
**Proposed changes**:
- Fixes https://github.com/RasaHQ/rasa/issues/8812

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] ~updated the documentation~
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: tests/utils/test_plotting.py [hunk 1]
```diff
@@ -29,6 +29,31 @@ def test_paired_histogram_specification_bins(
...
+            )
+
+
+@pytest.mark.parametrize(
+    "bad_data", [([[]]), ([[], []]),],
+)
+def test_plot_paired_histogram_doesnt_raise(bad_data: List):
+    """Empty data shouldn't raise an error."""
+    for density in [False, True]:
+        rasa.utils.plotting.plot_paired_histogram(bad_data, title="")
```
Comment (JEM-Mosig) [inline tests/utils/test_plotting.py:54]: I have to pass on the `density` parameter. Will do in a moment.

File: tests/utils/test_plotting.py [hunk 2]
```diff
@@ -29,6 +29,31 @@ def test_paired_histogram_specification_bins(
...
+            )
+
+
+@pytest.mark.parametrize(
+    "bad_data", [([[]]), ([[], []]),],
+)
+def test_plot_paired_histogram_doesnt_raise(bad_data: List):
+    """Empty data shouldn't raise an error."""
+    for density in [False, True]:
+        rasa.utils.plotting.plot_paired_histogram(bad_data, title="", density=density)
```
Comment (twerkmeister) [inline tests/utils/test_plotting.py:54]: could test for the user warning https://docs.pytest.org/en/6.2.x/warnings.html#warns

**Review by twerkmeister — COMMENTED** [review]
Neat! What do you think of my suggestions?

File: rasa/utils/plotting.py [hunk 3]
```diff
@@ -244,13 +250,17 @@ def plot_paired_histogram(
...
+    try:
+        bins, tallies, x_ranges, y_range = _extract_paired_histogram_specification(
+            histogram_data,
+            num_bins,
+            density=density,
+            x_pad_fraction=x_pad_fraction,
+            y_pad_fraction=y_pad_fraction,
+        )
+    except ValueError as e:
+        rasa.shared.utils.io.raise_warning(f"Unable to plot paried histogram: {e}")
```
[the remark this answers is not in the document]
Comment (JEM-Mosig) [reply rasa/utils/plotting.py:262]: Good point! I'll add the title, as we might want to use this kind of plot for other purposes in the future.

**Review by twerkmeister — APPROVED** [review]
Lookin' good! 🚀

**Comment by dakshvar22** [comment]
@JEM-Mosig Since the original bug resulted in model regression tests failing, did we run model regression tests on this PR to ensure that they don't fail again?

**Comment by JEM-Mosig** [comment]
I ran them locally without errors.

**Pull request merged**